### PR TITLE
feat: Add python bindings for mock and prove

### PR DESCRIPTION
**Pull request by JSeam2**
Changes:
1. Add python bindings for mock and prove
2. Refactor the run args in the function signature into a py_run_args class for better usability.
3. Adds python type conversions for VarVisibility, CheckMode, TranscriptType, StrategyType
4. Created a to_var_visibility helper function

### CODE REVIEW

File: src/python.rs [hunk 1]
```diff
@@ -164,9 +186,176 @@ fn forward(
...
+    let logrows = run_args.logrows;
+    let check_mode = run_args.check_mode;
+    let data = prepare_data(data);
+    let visibility = run_args.to_var_visibility();
+
+    match data {
+        Ok(d) => {
+            let procmodel = Model::<Fr>::new(model, run_args, Mode::Prove, visibility);
+
+            match procmodel {
```
Comment (alexander-camuto) [inline src/python.rs:277]: Could we clean this up and replace each match with something like 
```rust 
 let procmodel = Model::<Fr>::new(model, run_args, Mode::Prove, visibility);
 // replace with 
  let procmodel = Model::<Fr>::new(model, run_args, Mode::Prove, visibility).map_err(|_| PyIOError::new_err("Failed to process model"))?; 
```

File: src/python.rs [hunk 2]
```diff
@@ -164,9 +186,176 @@ fn forward(
...
+    let run_args: RunArgs = py_run_args.unwrap_or_else(PyRunArgs::new).into();
+    let logrows = run_args.logrows;
+    let data = prepare_data(data);
+    let visibility = run_args.to_var_visibility();
+
+    match data {
+        Ok(d) => {
+            let procmodel = Model::<Fr>::new(model, run_args, Mode::Mock, visibility);
+
+            match procmodel {
```
Comment (alexander-camuto) [inline src/python.rs:209]: Could we clean this up and replace each match with something like 
```rust 
 let procmodel = Model::<Fr>::new(model, run_args, Mode::Prove, visibility);
 // replace with 
  let procmodel = Model::<Fr>::new(model, run_args, Mode::Prove, visibility).map_err(|_| PyIOError::new_err("Failed to process model"))?; 
```

**Review by alexander-camuto — CHANGES_REQUESTED** [review]
LFG ! Added a few comments to reduce the pyramid of matches and keep things clean :)

File: src/python.rs (hunk 2 above)
Comment (JSeam2) [reply src/python.rs:209]: Let me try this, will get back soon!

**Review by alexander-camuto — APPROVED** [review]
SEND IT

**Pull request merged**